Adding tab support for triple pipes in "FOR" block

cryptoman256 opened the pull request:
This is an add on to the tab actions when editing.  Currently, when pressing "TAB" while editing in a test case or keyword table, you get a pipe ("|") added to the line and is limited to double pipes.

However, if you are in a ":FOR" block, the behavior is the same even though you need triples pipes.  This addresses that with some simple checks of ":FOR" or "| | |" in the previous line.  Triple pipe continuation was also added, ie. ("| | | ... |").

Commit:
Adding tab support for triple pipes in "FOR" block

boakley commented:
Thanks for the patch, matt! I'll see if I can find the time to integrate it this weekend.

mattplotner commented:
Great! I tested it locally and everything looked good.  Hope everything is
still going well with you!

On Thu, Sep 24, 2015 at 2:18 PM, Bryan Oakley <notifications@github.com>
wrote:

> Thanks for the patch, matt! I'll see if I can find the time to integrate
> it this weekend.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/boakley/brackets-robotframework/pull/28#issuecomment-143023896>
> .
>

boakley commented:
Matt, sorry this took me so long! I really appreciate the contribution.

cryptoman256 commented:
No problem, hope things are going well with you still.  You might be interested to know that Charlie and I refactored the page objects framework to work with Robot 2.9.2.  We stripped out some of the things that we don't care about and changed some other stuff like the robot alias decorator to use the @keyword one from Robot so we can have variables in the keyword name.  It's working pretty well so far.  I just need to clean up a few things.  Talk to you soon.

-Matt

Sent from my iPhone

> On Dec 26, 2015, at 5:15 PM, Bryan Oakley notifications@github.com wrote:
> 
> Matt, sorry this took me so long! I really appreciate the contribution.
> 
> —
> Reply to this email directly or view it on GitHub.

boakley commented:
Hah, It's funny you mention refactoring the page object stuff. I wrote my
own page object library and have been working on (but haven't finished) a
blog post about it. I've got the core page object library and PageObject
base class combined down to a total of only 117 lines of commented code
(compared to over 2000 lines of the robotframework-pageobject one we
started to use at Echo). It's a pretty sweet little library. When it gets
right down to it, you don't need much to implement page objects in robot.

If you're interested, I can tar up the code and give you some quick and
dirty instructions for trying it out. It might be a major effort for you to
switch over to it because I didn't try to be compatible with the other
library at all. It wouldn't be an insurmountable hurdle by any stretch, but
it doesn't use decorators, and doesn't inherit from the selenium2library so
you would have to change calls like "self.page_should_contain(...)" to
"self.se2lib.page_should_contain(...)".

Hope things are going well. Has IT settled into a new groove yet? Any
chance they are going to make you a full time employee in the new year?

--bryan

On Sat, Dec 26, 2015 at 6:01 PM, Matt Plotner notifications@github.com
wrote:

> No problem, hope things are going well with you still. You might be
> interested to know that Charlie and I refactored the page objects framework
> to work with Robot 2.9.2. We stripped out some of the things that we don't
> care about and changed some other stuff like the robot alias decorator to
> use the @keyword one from Robot so we can have variables in the keyword
> name. It's working pretty well so far. I just need to clean up a few
> things. Talk to you soon.
> 
> -Matt
> 
> Sent from my iPhone
> 
> > On Dec 26, 2015, at 5:15 PM, Bryan Oakley notifications@github.com
> > wrote:
> > 
> > Matt, sorry this took me so long! I really appreciate the contribution.
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/boakley/brackets-robotframework/pull/28#issuecomment-167372153
> .

cryptoman256 commented:
That's cool!  Ya, I forgot to mention that.  I'll be starting full time on Jan 4.  It was a long process but is finally happening.  

I would definitely like to check out what you've been working on.  We kept the Selenium inherit but destroyed base.  I did think about pulling out S2L though.

IT is still kind of a mess in my opinion.  They got the budget and project priorities finalized but it still feels like everyone is on different pages (no pun intended).  Not much has been accomplished either.  We've been helping with a SQL Server 2014 upgrade but that failed earlier this month and was postponed to January.  DevOps has been down to just Mike H for over a month now and so their projects have stalled since there's no bandwidth.  It's definitely still a work in progress but hopefully we keep moving forward.

-Matt

Sent from my iPhone

> On Dec 26, 2015, at 6:42 PM, Bryan Oakley notifications@github.com wrote:
> 
> Hah, It's funny you mention refactoring the page object stuff. I wrote my
> own page object library and have been working on (but haven't finished) a
> blog post about it. I've got the core page object library and PageObject
> base class combined down to a total of only 117 lines of commented code
> (compared to over 2000 lines of the robotframework-pageobject one we
> started to use at Echo). It's a pretty sweet little library. When it gets
> right down to it, you don't need much to implement page objects in robot.
> 
> If you're interested, I can tar up the code and give you some quick and
> dirty instructions for trying it out. It might be a major effort for you to
> switch over to it because I didn't try to be compatible with the other
> library at all. It wouldn't be an insurmountable hurdle by any stretch, but
> it doesn't use decorators, and doesn't inherit from the selenium2library so
> you would have to change calls like "self.page_should_contain(...)" to
> "self.se2lib.page_should_contain(...)".
> 
> Hope things are going well. Has IT settled into a new groove yet? Any
> chance they are going to make you a full time employee in the new year?
> 
> --bryan
> 
> On Sat, Dec 26, 2015 at 6:01 PM, Matt Plotner notifications@github.com
> wrote:
> 
> > No problem, hope things are going well with you still. You might be
> > interested to know that Charlie and I refactored the page objects framework
> > to work with Robot 2.9.2. We stripped out some of the things that we don't
> > care about and changed some other stuff like the robot alias decorator to
> > use the @keyword one from Robot so we can have variables in the keyword
> > name. It's working pretty well so far. I just need to clean up a few
> > things. Talk to you soon.
> > 
> > -Matt
> > 
> > Sent from my iPhone
> > 
> > > On Dec 26, 2015, at 5:15 PM, Bryan Oakley notifications@github.com
> > > wrote:
> > > 
> > > Matt, sorry this took me so long! I really appreciate the contribution.
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/boakley/brackets-robotframework/pull/28#issuecomment-167372153
> > .
> > 
> > —
> > Reply to this email directly or view it on GitHub.
